utils/src/workspace_path.rs

```diff
@@ -0,0 +1,35 @@
+// Copyright (c) 2021-2024 RBB S.r.l
+// opensource@mintlayer.org
+// SPDX-License-Identifier: MIT
+// Licensed under the MIT License;
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// https://github.com/mintlayer/mintlayer-core/blob/master/LICENSE
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/// Return the path to the workspace, as it was recorded on the build machine. The path has
+/// a trailing path separator.
+///
+/// Note that the path is produced by the build script using PathBuf::to_string_lossy, so if it
+/// contained non-Unicode characters, this value will be an approximation rather than the exact
+/// path itself.
+pub fn workspace_path() -> &'static str {
+    env!("WORKSPACE_PATH")
```

Comment:
Hmm so `CARGO_MANIFEST_DIR` is not accessible here directly?

Reply:
> Hmm so CARGO_MANIFEST_DIR is not accessible here directly?

Well, it is accessible, it's just that it contains the "/utils" suffix, so I'd have to make a lazy_static variable and remove the suffix at runtime. When I was writing this it seemed cleaner to do this in a build script. Right now I'm not sure if it's an overkill though.

Reply:
Hmm, ok. I usually prefer to avoid build scripts but don't have a very strong opinion in this case. Maybe instead of `lazy_static`, a function that strips whatever suffix is necessary every time would work too.

Reply:
> Maybe instead of lazy_static, a function that strips whatever suffix is necessary every time would work too.

I'm not opposing lazy_static per se, I just don't like the extra runtime work that is always the same. So, a function that does it every time is not better.

I gave this a thought once more and in the end I think that the build script approach is cleaner. So, since you don't insist on the alternative approach, I'm planning to leave it this way.
In the meantime, I've updated some comments and added a basic test.